sktime/forecasting/model_selection/_tune.py

```diff
@@ -267,7 +267,10 @@ def evaluate_candidates(candidate_params):
         # Select n best forecaster
         self.n_best_forecasters_ = []
         self.n_best_scores_ = []
-        for i in range(self.return_n_best_forecasters):
```

Comment:
I am wondering about doing

```python

if n == -1:
    self.return_n_best_forecasters = len(results.index)
```
But it is true that you are modifying an attribute on the fly
@fkiraly ?